realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -307,6 +307,27 @@ private boolean categorizeClassElements() {
                     }
                 }
 
+                if (variableElement.getAnnotation(Required.class) == null) {
+                    // The field doesn't have the @Required annotation
+                    if (!Utils.isPrimitiveType(variableElement) &&
```

Comment:
@cmelchior `isNullable()` is checking if a given field is nullable by looking for it in the `nullableElements`. And here is during the meta info scan and try to add the field into `nullableElements`.